### PR TITLE
Tests: Tendermint protocol dialogues

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

Tests for Tendermint protocol dialogues

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: packages/valory/protocols/tendermint/tests/test_tendermint.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    actual_envelope = Envelope.decode(envelope_bytes)
+    expected_envelope = envelope
+    assert expected_envelope.to == actual_envelope.to
+    assert expected_envelope.sender == actual_envelope.sender
+    assert (
+        expected_envelope.protocol_specification_id
+        == actual_envelope.protocol_specification_id
+    )
+    assert expected_envelope.message != actual_envelope.message
```
Comment (Adamantios) [inline packages/valory/protocols/tendermint/tests/test_tendermint.py:76]: Is there an issue for this?

```suggestion
    assert expected_envelope.message == actual_envelope.message
```

Comment (Karrenbelt) [reply packages/valory/protocols/tendermint/tests/test_tendermint.py:76]: https://github.com/valory-xyz/open-aea/issues/495

**Comment by Karrenbelt** [comment]
- `autonomy test --cov --cov-output test_results/tendermint_protocol by-path ./packages/valory/protocols/tendermint/`
    ```
    ---------- coverage: platform linux, python 3.10.4-final-0 -----------
    Name                                                     Stmts   Miss  Cover
    ----------------------------------------------------------------------------
    packages/valory/protocols/tendermint/__init__.py             4      4     0%
    packages/valory/protocols/tendermint/custom_types.py        12      9    25%
    packages/valory/protocols/tendermint/dialogues.py           23      0   100%
    packages/valory/protocols/tendermint/message.py             97     53    45%
    packages/valory/protocols/tendermint/serialization.py       77     20    74%
    packages/valory/protocols/tendermint/tendermint_pb2.py      40     40     0%
    ----------------------------------------------------------------------------
    TOTAL                                                      253    126    50%
    ```
- `pytest ./packages/valory/protocols/tendermint/tests/ --cov ./packages/valory/protocols/tendermint/ --cov-report term-missing`
    ```
    ---------- coverage: platform linux, python 3.10.4-final-0 -----------
    Name                                                            Stmts   Miss  Cover   Missing
    ---------------------------------------------------------------------------------------------
    packages/valory/protocols/tendermint/__init__.py                    4      0   100%
    packages/valory/protocols/tendermint/custom_types.py               12      0   100%
    packages/valory/protocols/tendermint/dialogues.py                  23      0   100%
    packages/valory/protocols/tendermint/message.py                    97      9    91%   156, 207-209, 244-250, 273-275
    packages/valory/protocols/tendermint/serialization.py              77      7    91%   60-62, 79, 111-112, 126
    packages/valory/protocols/tendermint/tendermint_pb2.py             22      0   100%
    packages/valory/protocols/tendermint/tests/__init__.py              1      0   100%
    packages/valory/protocols/tendermint/tests/test_tendermint.py      48      1    98%   132
    ---------------------------------------------------------------------------------------------
    TOTAL                                                             284     17    94%
    ```

**Review by Adamantios — DISMISSED** [review]

**Comment by Karrenbelt** [comment]
it is done now
```
---------- coverage: platform linux, python 3.10.4-final-0 -----------
Name                                                            Stmts   Miss  Cover   Missing
---------------------------------------------------------------------------------------------
packages/valory/protocols/tendermint/__init__.py                    4      0   100%
packages/valory/protocols/tendermint/custom_types.py               12      0   100%
packages/valory/protocols/tendermint/dialogues.py                  23      0   100%
packages/valory/protocols/tendermint/message.py                    97      0   100%
packages/valory/protocols/tendermint/serialization.py              77      0   100%
packages/valory/protocols/tendermint/tendermint_pb2.py             22      0   100%
packages/valory/protocols/tendermint/tests/__init__.py              1      0   100%
packages/valory/protocols/tendermint/tests/test_tendermint.py      62      1    98%   164
---------------------------------------------------------------------------------------------
TOTAL                                                             298      1    99%
```

File: packages/valory/protocols/tendermint/tests/test_tendermint.py [hunk 2]
```diff
@@ -95,7 +95,8 @@ def test_incorrect_error_data_logged(caplog: LogCaptureFixture) -> None:
         error_data=dict(message=1),  # incorrect type
     )
     expected = "Invalid type for dictionary values in content 'error_data'. Expected 'str'. Found '<class 'int'>'."
-    assert expected in caplog.text
+    with caplog.at_level(logging.ERROR, logger=_default_logger.name):
+        assert expected in caplog.text
```
Comment (Karrenbelt) [inline packages/valory/protocols/tendermint/tests/test_tendermint.py:102]: hopefully fixes the failure to capture logs seen on ubuntu 3.7 - 3.9 (but not 3.10) and on all macos runs, but none of the windows runs :upside_down_face:

File: packages/valory/protocols/tendermint/tests/test_tendermint.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 # ------------------------------------------------------------------------------
 #
-#   Copyright 2021 Valory AG
+#   Copyright 2022 Valory AG
```
Comment (Karrenbelt) [inline packages/valory/protocols/tendermint/tests/test_tendermint.py:4]: :eyes:

**Review by Adamantios — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**